### ISSUE TITLE
CKV_AWS_27 false-positive?

### ISSUE THREAD

**Issue by WestenMichael** [bridgecrewio/checkov#5061]
**Describe the issue**
Hello

I have a problem with CKV_AWS_27 rule. Checkov reporting that this is not supported in my terraform code but in terraform plan I can literally see that required property exists and have a value 

Here is what I have in terraform-plan file
```
{
  "address": "<prefix>.aws_sqs_queue.connection_change_events",
  "mode": "managed",
  "type": "aws_sqs_queue",
  "name": "connection_change_events",
  "provider_name": "registry.terraform.io/hashicorp/aws",
  "schema_version": 0,
  "values": {
    "arn": "arn:aws:sqs:eu-west-1:<account_id>:<prefix>-connection-change-events",
    "deduplication_scope": "",
    "delay_seconds": 0,
    "fifo_queue": false,
    "fifo_throughput_limit": "",
    "id": "<prefix>-connection-change-events",
    "kms_data_key_reuse_period_seconds": 300,
    "kms_master_key_id": "<some_resource_id>"
    <rest_parameters>
  }
}
```

What I'm doing wrong?

**Version (please complete the following information):**
 - Checkov Version 2.3.228 and 2.3.199

**Additional context**
This is not only this issue that looks like reporting false-positive

**Comment by nimrodkor** [comment]
Hello @WestenMichael !

Checkov scans on tfplan files the `resource_changes` part of the plan, which has a different structure. Can you share the dict for that part?

**Comment by WestenMichael** [comment]
Here is it:
```
{
  "address": "<prefix>.aws_sqs_queue.connection_change_events",
  "module_address": "<prefix>",
  "mode": "managed",
  "type": "aws_sqs_queue",
  "name": "connection_change_events",
  "provider_name": "registry.terraform.io/hashicorp/aws",
  "change": {
    "actions": [
      "update"
    ],
    "before": {
      "arn": "arn:aws:sqs:eu-west-1:<account_id>:<prefix>-connection-change-events",
      "content_based_deduplication": false,
      "deduplication_scope": "",
      "delay_seconds": 0,
      "fifo_queue": false,
      "fifo_throughput_limit": "",
      "id": "https://sqs.eu-west-1.amazonaws.com/<account_id>/<prefix>-connection-change-events",
      "kms_data_key_reuse_period_seconds": 3600,
      "kms_master_key_id": "<some_resource_id>",
      "max_message_size": 262144,
      <rest_params>
    },
    "after": {
      "arn": "arn:aws:sqs:eu-west-1:<account_id>:<prefix>-connection-change-events",
      "content_based_deduplication": false,
      "deduplication_scope": "",
      "delay_seconds": 0,
      "fifo_queue": false,
      "fifo_throughput_limit": "",
      "id": "https://sqs.eu-west-1.amazonaws.com/<account_id>/<prefix>-connection-change-events",
      "kms_data_key_reuse_period_seconds": 300,
      "kms_master_key_id": "<some_resource_id>",
      "max_message_size": 262144,
      <rest_params>
    },
    "after_unknown": {},
    "before_sensitive": {
      "tags": {},
      "tags_all": {}
    },
    "after_sensitive": {
      "tags": {},
      "tags_all": {}
    }
  }
}
```

**Comment by nimrodkor** [comment]
Alright, so on the plan file the check passes. I'm interested to see what you've done in code, as it might not be supported as a rendered variable path. Is it received from tfvars file? Can you share a little about how it gets populated? Is it hard coded?

**Comment by gruebel** [comment]
I just want to double check, you fail on scanning your `.tf` files, but a TF plan scan is successful? If yes, then it is probably related to us not being able to correctly evaluate conditions and/or function calls, as @nimrodkor mentioned.

Besides that I found an issue with the check, which could be related to your case, but doesn't has to be.

**Comment by WestenMichael** [comment]
@gruebel I failed scanning terraform-plan.json file (which was made from terraform output command file). Here are command that I used:
```
terraform plan -var-file=./config/dev/variables.tfvars -parallelism=30 -out=./terraform-plan
terraform show -json ./terraform-plan | jq '.' > ./terraform-plan.json
checkov --config-file .config.checkov.yml --quiet --file ./terraform-plan.json
```

@nimrodkor Here is how it looks like form Terraform code perspective
In main folder I have a file called `app.tf` that have a main module "device_mgmt". This module call another module in different folder
```
module "device_mgmt" {
  source = "./modules/device-mgmt"

  project                 = var.project
  env                     = var.env
  <rest parameters required by ./modules/device-mgmt>
}
```
`./modules/device-mgmt` is a folder with many modules. One of that modules is "async" module
```
module "async" {
  source = "../async"

  project    = var.project
  env        = var.env
  region     = var.region
  account_id = var.account_id

  schemas_version = var.schemas_version
  sqs_kms_key_id  = module.kms.sqs_kms_key_id
  sns_kms_key_id  = module.kms.sns_kms_key_id
  <rest_params>
}
```

In the same folder is also module "kms" and it looks like this:
```
module "kms" {
  source = "../kms"

  account_id      = data.aws_caller_identity.current.account_id
  project         = var.project
  env             = var.env
  whitelisted_roles = [
    <some_roles>
  ]
  tags = local.default_tags
}
```

`../kms` folder have a some files and one of them is sqs.tf that looks like this:
```
resource "aws_kms_key" "sqs" {
  description         = "KMS Key for SQS Queues"
  enable_key_rotation = true
  policy              = <<POLICY
  {
    ...
  }
  POLICY
}

resource "aws_kms_alias" "sqs" {
  name          = "alias/${var.project}-${var.env}-queues"
  target_key_id = aws_kms_key.sqs.key_id
}
```

And the last one. "async" module have a few files (one file per sqs) and each config for sqs contain this resource
```
resource "aws_sqs_queue" "requests" {
  name                      = "${var.project}-${var.env}-requests"
  kms_master_key_id                 = var.sqs_kms_key_id
  kms_data_key_reuse_period_seconds = 3600
  <rest_params>
}
```

**Comment by gruebel** [comment]
when you used Terraform plan, then I think my fix should actually resolve your problem. Please give the the latest version a try 🙂

**Comment by WestenMichael** [comment]
Thx for info @gruebel. I will check it today/tomorrow

**Comment by WestenMichael** [comment]
I can confirm the fix work correctly :) I will check with other rules, too.

Thx for help @gruebel @nimrodkor. Issue can be closed.
